theano/gof/op.py

```diff
@@ -779,10 +780,33 @@ class Op(utils.object2, PureOp, CLinkerOp):
     Convenience class to bundle `PureOp` and `CLinkerOp`.
 
     """
+    instances = {}
+
     def __new__(cls, *args, **kwargs):
         # this function exists to silently and transparently ensure that all
         # existing Ops get a _op_use_c_code attribute
-        obj = object.__new__(cls)
+        key = list()
+        if args:
+            [key.append(i) for i in args]
+        if kwargs:
+            [key.append(v) for k,v in kwargs.iteritems()]
```

Comment:
All the param passed to the __init__ will be recieved here as well. So, `(*args, **kwargs)` is a more generalized way. 

> We need to get the `__props__`.

Meaning, iterating through cls.__props__ ?